src/puppetlabs/puppetdb/scf/migrate.clj

```diff
@@ -890,11 +890,118 @@
+                      ["end_time"              "timestamp with time zone NOT NULL"]
+                      ["receive_time"          "timestamp with time zone NOT NULL"]
+                      ["transaction_uuid"      "varchar(255) DEFAULT NULL"]
+                      ["noop"                  "boolean"]
+                      ["environment_id"        "bigint"]
+                      ["status_id"             "bigint"])
+
+    (sql/create-table :resource_events_transform
+                      ["report_id"        "bigint NOT NULL"]
+                      ["status"           "varchar(40) NOT NULL"]
+                      ["timestamp"        "timestamp with time zone NOT NULL"]
+                      ["resource_type"    "text NOT NULL"]
+                      ["resource_title"   "text NOT NULL"]
+                      ["property"         "varchar (40)"]
+                      ["new_value"        "text"]
+                      ["old_value"        "text"]
+                      ["message"          "text"]
+                      ["file"             "varchar(1024) DEFAULT NULL"]
+                      ["line"             "integer"]
+                      ["containment_path" (scf-utils/sql-array-type-string "TEXT")]
+                      ["containing_class" "varchar(255)"])
+
+    (sql/create-table :latest_reports_transform
+                      ["certname"  "text NOT NULL"]
+                      ["report_id" "bigint NOT NULL"])
+
+
+    (sql/do-commands
+     "INSERT INTO reports_transform (
+       hash, certname, puppet_version, report_format, configuration_version,
+       start_time, end_time, receive_time, transaction_uuid, environment_id,
+       status_id)
+       SELECT hash, certname, puppet_version, report_format,
+         configuration_version, start_time, end_time, receive_time,
+         transaction_uuid, environment_id, status_id
+         FROM reports")
+
+    (sql/do-commands
+     "ALTER TABLE reports_transform
+       ADD CONSTRAINT reports_hash_key UNIQUE (hash)")
+
+    (sql/do-commands
+     "INSERT INTO resource_events_transform (
+        report_id, status, timestamp, resource_type, resource_title, property,
+        new_value, old_value, message, file, line, containment_path,
+        containing_class)
+        SELECT rt.id, status, timestamp, resource_type, resource_title,
+          property, new_value, old_value, message, file, line, containment_path,
+          containing_class
+          FROM resource_events AS re
+          INNER JOIN reports_transform rt on re.report = rt.hash")
+
+    (sql/do-commands
+     "INSERT INTO latest_reports_transform(certname, report_id)
+        SELECT lr.certname, rt.id
+          FROM latest_reports AS lr
+          INNER JOIN reports_transform rt on lr.report = rt.hash")
+
+    (sql/do-commands
+     "DROP TABLE resource_events"
+     "DROP TABLE latest_reports"
+     "DROP TABLE reports")
+
+    ;; TODO: do we need to delete other references here, or do they magically disappear?
```

Comment:
Another one of my TODO's :-). Objects related to these tables disappear, I think the main exception is the way we create sequences now. But since these objects don't use these sequences, doesn't matter in this case.